### PR TITLE
Make controllers/ golint compatible

### LINKED ISSUE #1055: Make controllers/ golint compatible
KEDA uses [golint](https://github.com/golang/lint) to ensure code quality. However, some files still need adjusting:

- [ ] controllers/scaledjob_controller.go
- [ ] controllers/scaledobject_controller.go
- [ ] controllers/util/status.go
- [ ] controllers/util/string_lists.go

### How 

1. Create a new branch from `v2`branch
2. Remove the files from list in [.pre-commit-config.yaml](https://github.com/kedacore/keda/blob/v2/.pre-commit-config.yaml)
3. Either run `pre-commit go-lint --all-files` (this requires installed [pre-commit](https://github.com/kedacore/keda/blob/v2/CONTRIBUTING.md#code-quality)) or `golint path/to/selected/file.go`
4. Commit your changes and open a PR to `v2`

### Reference
https://github.com/kedacore/keda/pull/1025#discussion_r475134492

### PR DESCRIPTION
**Pull request by Mystigan**
- [x] controllers/util/status.go : Comments added for exported functions: "SetStatusConditions" and "UpdateScaledObjectStatus".
- [x] controllers/util/string_lists.go : Comments added for exported functions: "Contains" and "Remove".
- [x] controllers/scaledjob_controller.go : Comments added for exported methods: "SetupWithManager" and "Reconcile". An else block readjusted to make it golint compliant.
- [x] controllers/scaledobject_controller.go : Comments added to exported methods: "SetupWithManager" and "Reconcile". Two else blocks readjusted to make it golint compliant.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [ ] Tests have been added
- [ ] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs
- [ ] Changelog has been updated

Fixes #1055

### CODE REVIEW

**Comment by Mystigan** [comment]
Hi folks,
This is my first time contributing to an open source project. I have done things to the best of my knowledge. Please let me know if something is missing or was done incorrectly.
Any and all constructive criticism is welcome.

File: controllers/scaledjob_controller.go [hunk 1]
```diff
@@ -41,6 +42,7 @@ func (r *ScaledJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+//Reconcile performs reonciliation on the identified Kubernetes job based on the request information passed, returns the result and an error (if any).
```
Comment (zroubalik) [inline controllers/scaledjob_controller.go:45]: I'd replace `Kubernetes job` with `ScaledJob resource`

File: controllers/scaledobject_controller.go [hunk 2]
```diff
@@ -100,6 +101,7 @@ func initScaleClient(mgr manager.Manager, clientset *discovery.DiscoveryClient)
 	), nil
 }
 
+//Reconcile performs reonciliation on the identified Kubernetes object based on the request information passed, returns the result and an error (if any).
```
Comment (zroubalik) [inline controllers/scaledobject_controller.go:104]: I'd replace `Kubernetes object` with `ScaledObject resource`

File: controllers/scaledjob_controller.go [hunk 3]
```diff
@@ -30,6 +30,7 @@ type ScaledJobReconciler struct {
 	scaleHandler scaling.ScaleHandler
 }
 
+//SetupWithManager initializes the ScaledJobReconciler instance and starts a new controller managed by the passed Manager instance.
```
Comment (zroubalik) [inline controllers/scaledjob_controller.go:33]: Could you please put a space behind the slash, so it is in sync with the rest of the comments in the code. ie:
`//SetupWithManager` -> `// SetupWithManager`

This applies to the other comments you have added as well

**Review by zroubalik — CHANGES_REQUESTED** [review]
Welcome to the opensource world! 🎉  Thanks for your contribution, this is really great. 

Looking good, I have just a minor nits.

File: controllers/scaledjob_controller.go (hunk 3 above)
Comment (Mystigan) [reply controllers/scaledjob_controller.go]: > Could you please put a space behind the slash, so it is in sync with the rest of the comments in the code. ie:
> `//SetupWithManager` -> `// SetupWithManager`
> 
> This applies to the other comments you have added as well

Updated comments to have a single space after the slash.

File: controllers/scaledjob_controller.go (hunk 1 above)
Comment (Mystigan) [reply controllers/scaledjob_controller.go]: > I'd replace `Kubernetes job` with `ScaledJob resource`

Marked words have been replaced.

File: controllers/scaledobject_controller.go (hunk 2 above)
Comment (Mystigan) [reply controllers/scaledobject_controller.go]: > I'd replace `Kubernetes object` with `ScaledObject resource`

I have replaced the words.

**Review by zroubalik — APPROVED** [review]
LGTM, thanks!

**Pull request merged**